docs/trace_collection.md

```diff
@@ -216,206 +216,206 @@ Before data is uploaded to Datadog, it is stored in cleartext in the cache direc
+This sets additional tracing headers on your request so your backend can extract the request and continue distributed tracing. Once the request is done, call `span.finish()` within a completion handler. If your backend is also instrumented with [Datadog APM & Distributed Tracing][10], the entire front-to-back trace appears in the Datadog dashboard.
 
-    NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration]
-                                                          delegate:[[DDNSURLSessionDelegate alloc] init]
-                                                     delegateQueue:nil];
-    ```
-    {{% /tab %}}
-    {{< /tabs >}}
+* In order for the SDK to automatically trace all network requests made to the given hosts, specify the `firstPartyHosts` array in the Datadog initialization and use `DDURLSessionDelegate` as a delegate of the `URLSession` instance you want to monitor:
+
+{{< tabs >}}
+{{% tab "Swift" %}}
+```swift
+Datadog.initialize(
+    appContext: .init(),
+    configuration: Datadog.Configuration
+        .builderUsing(clientToken: "<client_token>", environment: "<environment_name>")
+        .trackURLSession(firstPartyHosts: ["example.com", "api.yourdomain.com"])
+        .build()
+)
+
+let session = URLSession(
+    configuration: .default,
+    delegate: DDURLSessionDelegate(),
+    delegateQueue: nil
+)
+```
+{{% /tab %}}
+{{% tab "Objective-C" %}}
+```objective-c
+DDConfigurationBuilder *builder = [DDConfiguration builderWithClientToken:@"<client_token>"
+                                                                environment:@"<environment_name>"];
+
+[builder trackURLSessionWithFirstPartyHosts:[NSSet setWithArray:@[@"example.com", @"api.yourdomain.com"]]];
+
+[DDDatadog initializeWithAppContext:[DDAppContext new]
+                    trackingConsent:trackingConsent
+                        configuration:[builder build]];
+
+NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration]
+                                                        delegate:[[DDNSURLSessionDelegate alloc] init]
+                                                    delegateQueue:nil];
+```
+{{% /tab %}}
+{{< /tabs >}}
 
-    This will trace all requests made with this `session` to `example.com` and `api.yourdomain.com` hosts (for example, `https://api.yourdomain.com/v2/users` or `https://subdomain.example.com/image.png`).
+This traces all requests made with this `session` to `example.com` and `api.yourdomain.com` hosts (for example, `https://api.yourdomain.com/v2/users` or `https://subdomain.example.com/image.png`).
 
-    **Note**: Tracing auto instrumentation uses `URLSession` swizzling, but it is opt-in: if you do not specify `firstPartyHosts`, no swizzling is applied.
+**Note**: Tracing auto-instrumentation uses `URLSession` swizzling and is opt-in. If you do not specify `firstPartyHosts`, swizzling is not applied.
```

Comment:
Is "swizzling" defined elsewhere, or is it a commonly understood term? It is a bit confusing to me, but this is outside my normal subject area.

Reply:
Yes, is a commonly understood term in iOS programming. Anyway this change was not supposed to be here so I reverted it